src/node/stage/approved.rs

```diff
@@ -1561,6 +1561,16 @@ impl Approved {
             info!("Promoted to elder");
             core.send_event(Event::PromotedToElder);
 
+            // For the case that adult got promoted into elder via update, EldersInfo needs to be
+            // inserted into dkg_voter manually. So that the voted dkg_result can be fetched and
+            // update section_key_provider properly.
+            let info = self.shared_state.our_info().clone();
+            let participants: BTreeSet<_> = info.elder_ids().copied().collect();
+            let section_key_index = self.shared_state.our_history.last_key_index() - 1;
+            let dkg_key = (participants, section_key_index);
+            trace!("inserting dkg info {:?} {:?}", dkg_key, info);
+            self.dkg_voter.insert_info(&dkg_key, info);
```

Comment:
Could you explain why is this necessary? If I understand it correctly, it's for the case where a node receives `Sync` with the new section info before its DKG completes. But isn't that already covered by this: https://github.com/maidsafe/routing/blob/master/src/node/stage/approved.rs#L1353-L1360 ? Or is it something else?

Reply:
This is mainly for the case that ALL elders got relocated, i.e. all new elders are previous adults.
In which, those adults doesn't have the new_elders_info to populate the dkg_voter, hence no one will vote for the OurKey and SectionInfo.

The code block you pointing to doesn't work in this case, as the shared_state updated via sync doesn't contain this section_key of all new_elders.

Reply:
Ok. It seems due to the issue as address by [only call promote_demote_elders when elders_info updated](https://github.com/maidsafe/routing/pull/2188/commits/d2aa569b7ef28d1a5f3e770812d164ffd9565aeb) , there is un-necessary additional increase of section_index, which cause the shared_state doesn't updated properly.

I tried with revert this change [insert EldersInfo when being updated](https://github.com/maidsafe/routing/pull/2188/commits/4f7428b832f3cf0e1d613b859bc018cdb28c375f), which still passing all unit tests and most of the integration tests.

So, as long as there is no un-necessary extra votes, this change is not required.

I will update the PR with this commit removed.